File: twoSum.js
const twoSum = (nums, target) => {
	let previousValues = {};

	for (let i = 0; i < nums.length; i++) {
		let currentNumber = nums[i];
		let neededValue = target - currentNumber;
		let index2 = previousValues[neededValue]
		if (index2 != null) {
			return [index2, i]
		} else {
			previousValues[currentNumber] = i;
		}		
	}
}

twoSum([2, 7, 11, 15], 9)

File: two_sum.py
#! python3

def twoSum(nums, target):
    prevMap = {} #val : index
    for index, value in enumerate(nums):
        diff = target - value
        if diff in prevMap:
            print(prevMap[diff], index)
        else:
            prevMap[value] = index

twoSum([2,7,11,15], 9)

